datagen/src/main/scala/com/nvidia/rapids/tests/scaletest/TableGenerator.scala

```diff
@@ -0,0 +1,231 @@
+    - complexity data columns
+   */
+  private def genAFacts(): DataFrame = {
+    val schema = "primary_a long," +
+      "a_key4_1 int," +
+      "a_data_low_unique_1 long," +
+      "a_data_low_unique_len_1 string," +
+      (1 to complexity).map(i => s"a_data_$i ${randomColumnType()}").mkString(",")
+    val aFacts = dbgen.addTable("a_facts", schema, aNumRows)
+    aFacts("primary_a").setSeedRange(1, aNumRows)
+    aFacts("a_key4_1").setSeedRange(1,5)
+    aFacts("a_data_low_unique_1").setSeedRange(1, 5)
+    aFacts("a_data_low_unique_len_1").setLength(1, 5)
+    // ride-along columns will all use default data.
+    aFacts.toDF(spark)
+  }
+
+  /**
+  b_data: Each scale factor corresponds to 1,000,000 rows. (The row group size should be configured to be 512 MiB when writing parquet or equivalent when writing ORC)
+    - b_foreign_a Should overlap with a_facts.primary_a about 99% of the time
+    - key group 3 with 3 columns (The unique count should be about 99% of the total number of rows).
+    - 10 data columns of various types
+    - 10 decimal(10, 2) columns with a relatively small value range
+   */
+  private def genBData(): DataFrame = {
+    val schema = "b_foreign_a long," +
+      "b_key3_1 string," +
+      "b_key3_2 date," +
+      "b_key3_3 long," +
+      (1 to 10).map(i => s"b_data_$i ${randomColumnType()}").mkString(", ") + ", " +
+      (1 to 10).map(i => s"b_data_small_value_range_$i decimal(10, 2)").mkString(", ")
+    val overlapRatio = 0.99
+    val bData = dbgen.addTable("b_data", schema, bNumRows)
+    // This column "Should overlap with a_facts.primary_a about 99% of the time"
+    bData("b_foreign_a").setSeedRange(1, Math.round(1 / overlapRatio * aNumRows))
+    (1 to 3).map(i => bData(s"b_key3_$i").setSeedRange(1, Math.round(bNumRows * 0.99)))
+    bData("b_key3_2").setValueRange(-719162, 2932896)
+    (1 to 10).map(i => bData(s"b_data_small_value_range_$i").setSeedRange(1, 10000))
+    bData.toDF(spark)
+  }
+
+  /**
+  c_data: Each scale factor corresponds to 100,000 rows.
+    - c_foreign_a Should overlap with a_facts.primary_a about 50% of the time
+    - key group 2 columns up to the complexity (each key should show up about 10 times)
+    - c_data_row_num_1 1 data column that is essentially the row number for order by tests in window
+    - 5 data columns
+    - 5 numeric data columns
+   */
+  private def genCData(): DataFrame = {
+    val schema = "c_foreign_a long," +
+      (1 to complexity).map(i => s"c_key2_$i ${randomColumnType()}").mkString(",") + "," +
+      "c_data_row_num_1 long," +
+      (1 to 5).map(i => s"c_data_$i ${randomColumnType()}").mkString(",") + "," +
+      (1 to 5).map(i => s"c_data_numeric_$i ${randomNumericColumnType()}").mkString(",")
+    val overlapRatio = 0.50
+    val cData = dbgen.addTable("c_data", schema, cNumRows)
+    cData("c_foreign_a").setSeedRange(1, Math.round(1 / overlapRatio * aNumRows))
```

Comment:
Again I thought that overlap meant that 50% of the keys here would match with the keys in `primary_a`, which means 50% of them would not be in `primary_a`. Not that the keys here would only cover half of the key range of `primary_a`